riscv-sbi.adoc

```diff
@@ -257,6 +258,85 @@ call doesn't return.
 | *RESERVED*                |             |    0x09-0x0F |
 |===
 
+== Hart Hotplug Extension, Extension ID: 0x484850 (HHP)
+
+The Hart Hotplug Extension introduces a set of functions that allow the supervisor
+software to request M-mode to add/remove harts to the list of available hart to
+the supervisor.
+
+[source, C]
+----
+struct sbiret sbi_hart_add(unsigned long hartid, unsigned long start_addr, unsigned
+long priv)
+----
+
+Brings up *hartid* either during initial boot or after an `sbi_hart_remove` SBI
+call. This will be an asynchronous call. That means the caller on the supervisor
+side need to verify if the cpu is really up or not.
+
+*start_addr* points to a runtime-specified physical address, where a hart can
+resume execution after its initialization/resume sequence.  Before jumping to
+*start_addr*, the hart MUST configure PMP if present and switch to Supervisor
+mode.
+
+*priv* is a pointer to a opaque data segment that the caller can use to pass
+information about execution context.
+
+*Returns* one of the following possible SBI error code thorugh sbiret.error.
+
+* SBI_SUCCESS
+* SBI_ERR_INVALID_ADDRESS
+* SBI_ERR_INVALID_PARAM
+* SBI_ERR_ALREADY_AVAILABLE
+* SBI_ERR_FAILED
+
+[source, C]
+----
+struct sbiret sbi_hart_remove()
+----
+
+Mark the calling hart unavailable to the supervisor mode software. The hart can only
+be made available through the `sbi_hart_add` call. This is a synchronous call and
+is not expected to return if succeeds.
```

Comment:
Please add a statement that a hart can only remove itself.

I know it is implicit from sbi_hart_remove() function prototype but we should have an explicit statement here to avoid confusion.